jujugui/static/gui/src/app/components/copy-to-clipboard/copy-to-clipboard.js

```diff
@@ -0,0 +1,77 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2015 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('copy-to-clipboard', function() {
+
+  juju.components.CopyToClipboard = React.createClass({
+
+    clipboard: null,
+
+    /* Define and validate the properites available on this component. */
+    propTypes: {
+      value: React.PropTypes.string
+    },
+
+    getDefaultProps: function() {
+      return {
+        className: 'copy-to-clipboard',
```

Comment:
Discussed with @hatched and decided to leave as is; there's no logic around building the class name (i.e., conditionally concatenating). The only reason for this is to let the parent component set the block and then assign elements within the block the correct classname, according to BEM standards.